modin/pandas/test/dataframe/test_window.py

```diff
@@ -243,22 +242,62 @@ def test_fillna_inplace():
 
 
 @pytest.mark.parametrize("data", test_data_values, ids=test_data_keys)
-def test_frame_fillna_limit(data):
+@pytest.mark.parametrize("limit", [1, 2, 0.5, -1, -2, 1.5])
+def test_frame_fillna_limit(data, limit):
     pandas_df = pandas.DataFrame(data)
 
-    index = pandas_df.index
+    replace_pandas_series = pandas_df.columns.to_series().sample(frac=1)
+    replace_dict = replace_pandas_series.to_dict()
+    replace_pandas_df = pandas.DataFrame(
+        {col: pandas_df.index.to_series() for col in pandas_df.columns},
+        index=pandas_df.index,
+    ).sample(frac=1)
+    replace_modin_series = pd.Series(replace_pandas_series)
+    replace_modin_df = pd.DataFrame(replace_pandas_df)
 
+    index = pandas_df.index
     result = pandas_df[:2].reindex(index)
     modin_df = pd.DataFrame(result)
+
+    if isinstance(limit, float):
+        limit = int(len(modin_df) * limit)
+    elif limit < 0:
```

Comment:
what would happen if `limit` is `-0.5`?.. suggested fix:
```suggestion
    if limit < 0:
```

Reply:
Ok.